### PR TITLE
Attachment shortcuts support on native devices

### PR DESCRIPTION
**Pull request by azimgd**
### Fixed Issues

$ https://github.com/Expensify/App/issues/18268
PROPOSAL: https://github.com/Expensify/App/issues/18268

### Tests

- [x] Verify that no errors appear in the JS console

1. Connect hardware keyboard
2. Open any chat and press "+" icon in composer, select "Add Attachment"
3. Arrow up / Arrow down should scroll through the list
4. Open a chat with multiple attachments and click on any recent one
5. Arrow left / Arrow Right should scroll through attachments

### Offline tests

Same as tests

### QA Steps

Same as tests

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/4882133/236507610-7bf70165-1492-46d5-88b0-200b734429ac.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/4882133/236513430-924a3817-3b20-4a26-89b0-da231af686ad.mov

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/4882133/236507182-d454a123-5b8a-4094-9482-65601964885f.mov

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/4882133/236510136-544a0b00-f74f-4563-8e24-be6bd2acf656.mov

</details>

<details>
<summary>iOS</summary>

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/4882133/236512053-a44e7f74-fee3-4050-a8f9-59a0e4c7a12a.MOV

</details>

### CODE REVIEW

**Comment by azimgd** [comment]
@rushatgabhane the code is ready to be reviewed. I will complete the checklist and add screenshots shortly.

**Comment by azimgd** [comment]
@rushatgabhane ready.

**Comment by azimgd** [comment]
@rushatgabhane ready.

**Review by rushatgabhane — CHANGES_REQUESTED** [review]
@azimgd code looks good to me! and tests well on android 

left a small change request to use functional component.

**Comment by luacmartins** [comment]
@azimgd conflicts

File: src/components/AttachmentCarousel/index.js [hunk 1]
```diff
@@ -219,7 +219,7 @@ class AttachmentCarousel extends React.Component {
         const nextIndex = this.state.page - deltaSlide;
         const nextItem = this.state.attachments[nextIndex];
 
-        if (!nextItem) {
+        if (!nextItem || !this.scrollRef.current) {
```
[the remark this answers is not in the document]
Comment (azimgd) [reply src/components/AttachmentCarousel/index.js:212]: I assume that was some sort of a race condition happening on Android where scrollRef was undefined upon scroll.

**Comment by azimgd** [comment]
updated.

**Review by luacmartins — DISMISSED** [review]
Looking good. We just need to address @rushatgabhane's comment.

**Comment by azimgd** [comment]
Hey @rushatgabhane would you mind taking a look one more time please. @luacmartins I haven't implemented functional component initially as the `CarouselActions/index.js` was implemented as a class component, so I followed the same pattern when created a new file.

**Comment by azimgd** [comment]
@rushatgabhane are we good to add `// eslint-disable-next-line react-hooks/exhaustive-deps` as these effects don't have a "changing" dependencies?

**Review by rushatgabhane — APPROVED** [review]
@luacmartins LGTM! 🚀

**Review by luacmartins — APPROVED** [review]
LGTM

**Pull request merged**